### DISCUSSION TITLE
Temporary File Uploads with DigitalOcean Spaces

### DISCUSSION THREAD

**Discussion by dennisprudlo** [livewire/livewire#4594] [1. Help] [answered]
Hey guys,
I recently implemented basic file uploads with livewire and everything works perfectly. The temporary file is stored in the storage and then finally stored in the DigitalOcean Space in a specific folder.

Now I wanted to configure the file upload that the temporary file is uploaded to the space directly without being stored on my server. I configured livewire to use the s3 disk and apparently livewire tries to upload the file to the Space (PUT request to
https://xyz.digitaloceanspaces.com/livewire-tmp). However the request fails with `XMLHttpRequest cannot load {url} due to access control checks` and `Failed to load resource: Preflight response is not successful`.

Are DigitalOcean Spaces not supported for temporary file uploads or is there anything else that needs to be configured first? The actual final upload to the Space works its just the temporary file upload that doesn't work.

**Comment by dennisprudlo** [comment, marked as answer]
OK trying in chrome I get the info that this is a CORS issue, safari didn't said anything about CORS so that might be the problem. Will check my CORS Headers later. But this seems solvable 👍🏼

**Reply by dennisprudlo** [reply]
Ok found the issue: DigitalOcean by default does not allow CORS request, they need to be configured first. Then all non-default Headers need to be allowed. So I allowed "x-amz-acl" and it still didn't work.

Well turns out that "Content-Type" is a default header you don't have to allow in your CORS-config except the content type is different from `application/x-www-form-urlencoded`, `multipart/form-data` or `text/plain` (https://developer.mozilla.org/en-US/docs/Web/HTTP/CORS#simple_requests).

And since we try to upload images here the solution was to add "x-amz-acl" and "Content-Type" as Allowed Headers to the Spaces CORS configuration.

@calebporzio Maybe something for the docs? I don't know about Amazon S3 but apparently you don't have to do that by yourself. Maybe it is preconfigured to allow these headers but it took quite some time to finally get it working without allowing all headers.

**Reply by joshhanley** [reply]
@dennisprudlo glad you sorted it out! PR's to the docs are welcome https://github.com/livewire/docs

**Reply by dennisprudlo** [reply]
@joshhanley PR'd 👍🏼

**Reply by hashemirafsan** [reply]
@dennisprudlo you saved me 👍

**Reply by maurocasas** [reply]
🐐

**Reply by mashuble** [reply]
🐐

**Comment by hasan-aftab** [comment]
Still facing same error

**Reply by iamsandakelum** [reply]
This time it's because of a undocumented breaking change on Livewire v4.0+ at https://github.com/livewire/livewire/discussions/9774
Is it the same exact issue you're having?
